sympy/polys/numberfields/primes.py

```diff
@@ -0,0 +1,490 @@
+"""Prime ideals in number fields. """
+
+from sympy import symbols
+from sympy.polys import Poly
+from sympy.polys.domains import FF, QQ, ZZ
+from sympy.polys.matrices import DomainMatrix
+from sympy.polys.polyerrors import CoercionFailed
+from sympy.polys.polyutils import IntegerPowerable
+from sympy.utilities import public
+
+from .basis import round_two, nilradical_mod_p
+from .modules import ModuleEndomorphism, Order, find_min_poly
+from .utilities import coeff_search, supplement_a_subspace
+
+
+class PrimeIdeal(IntegerPowerable):
+    """
+    A prime ideal in a ring of algebraic integers.
+    """
+
+    def __init__(self, p, alpha, e, f, ZK):
+        """
+        Parameters
+        ==========
+
+        p: the rational prime this ideal divides.
+        alpha: :py:class:`PowerBasisElement` such that the ideal is equal to ``p*ZK + alpha*ZK``.
+        e: the ramification index.
+        f: the inertia degree.
+        ZK: the :py:class:`Order` where this ideal lives
+
+        """
+        self.p = p
+        self.alpha = alpha
+        self.e = e
+        self.f = f
+        self.ZK = ZK
+        self._beta = None
+
+    def pretty(self, theta=None):
+        theta = theta or symbols('x')
+        p, alpha, e, f = self.p, self.alpha, self.e, self.f
+        alpha_rep = str(alpha.numerator(x=theta).as_expr())
+        if alpha.denom > 1:
+            alpha_rep = f'({alpha_rep})/{alpha.denom}'
+        return f'[ ({p}, {alpha_rep}) e={e}, f={f} ]'
+
+    def __repr__(self):
+        return self.pretty()
+
+    def as_ideal(self):
+        return self.p * self.ZK + self.alpha * self.ZK
+
+    def copy(self):
+        return PrimeIdeal(self.p, self.alpha.copy(), self.e, self.f, self.ZK.copy())
```

Comment:
There are a few `copy` methods on different classes in this PR. As far as I can tell all of these classes are immutable so it isn't really meaningful to make copies.

Reply:
Agreed. Copying is gone now.